intermine_boot/intermine_docker.py

```diff
@@ -0,0 +1,300 @@
+        else:
+            print ('Configuration change detected. Downloading compose file...')
+            shutil.rmtree(compose_path.parent)
+    
+    if not same_conf_exist:
+        print ('Same conf not found...', compose_path.parent)
+        Repo.clone_from(DOCKER_COMPOSE_REPO, compose_path.parent, 
+                    progress=utils.GitProgressPrinter())
+
+    _create_volume_dirs(compose_path)
+
+    option_vars = (['IM_REPO_URL='+options['im_repo'],
+                    'IM_REPO_BRANCH='+options['im_branch']]
+                   if options['build_im'] else [])
+
+    client = docker.from_env()
+    if options['build_images']:
+        print ('Building images...')
+        img_path = compose_path.parent
+        tomcat_image = client.images.build(
+            path=str(img_path / 'tomcat'), dockerfile='tomcat.Dockerfile')
+        solr_image = client.images.build(
+            path=str(img_path / 'solr'), dockerfile='solr.Dockerfile')
+        postgres_image = client.images.build(
+            path=str(img_path / 'postgres'), dockerfile='postgres.Dockerfile')
+        intermine_builder_path = client.images.build(
+            path=str(img_path / 'intermine_builder'), dockerfile='intermine_builder.Dockerfile')
+    else:
+        print ('Pulling images...')
+        tomcat_image = client.images.pull('intermine/tomcat:latest')
+        solr_image = client.images.pull('intermine/solr:latest')
+        postgres_image = client.images.pull('intermine/postgres:latest')
+        intermine_builder_image = client.images.pull('intermine/builder:latest')
+
+    print ('Starting containers...')
+    tomcat_container = create_tomcat_container(client, tomcat_image)
+    solr_container = create_solr_container(client, solr_image, env)
+    postgres_container = create_postgres_container(client, postgres_image, env)
+    intermine_builder_container = create_intermine_builder_container(
+        client, intermine_builder_image, env)
+        # # Make sure dockerhub images are up-to-date.
+        # subprocess.run(['docker-compose',
+        #                 '-f', compose_path.name,
+        #                 'pull'],
+        #                check=True,
+        #                cwd=compose_path.parent)
+
+
+    # subprocess.run([*ENV_VARS, *option_vars,
+    #                 'docker-compose',
+    #                 '-f', compose_path.name,
+    #                 'up', '-d'] +
+    #                (['--build', '--force-recreate']
+    #                 if options['build_images'] else []),
+    #                check=True,
+    #                cwd=compose_path.parent)
+    
+    _store_conf(env['data_dir'], options)
+
+
+def down(options, env):
+    compose_path = _get_compose_path(options, env)
+    subprocess.run([*ENV_VARS,
+                    'docker-compose',
+                    '-f', compose_path.name,
+                    'down'],
+                   check=True,
+                   cwd=compose_path.parent)
```

Comment:
Why we are still using docker-compose to stop containers? Is it still work in progress?

Reply:
yes

Reply:
It's still in progress. Hence even commented code is there of compose